File: ros2_ws/src/lifecycle_pkg/lifecycle_pkg/my_talker.py
#!/usr/bin/env python3
import rclpy
from rclpy.lifecycle import LifecycleNode
from rclpy.lifecycle.node import LifecycleState, TransitionCallbackReturn
from example_interfaces.msg import String

class MyTalkerNode(LifecycleNode):
    def __init__(self):
        super().__init__("my_talker")
        self.get_logger().info("Executing constructor")
        self.counter_ = 0

    def on_configure(self, previous_state: LifecycleState):
        self.get_logger().info(f"Node '{self.get_name()}' is in state '{previous_state.label}', executing on_configure")
        self.publisher_ = self.create_lifecycle_publisher(String, "my_chatter", 10)
        self.timer_ = self.create_timer(0.5, self.publish_talker)
        return TransitionCallbackReturn.SUCCESS

    def on_cleanup(self, previous_state: LifecycleState):
        self.get_logger().info(f"Node '{self.get_name()}' is in state '{previous_state.label}', executing on_cleanup")
        self.destroy_lifecycle_publisher(self.publisher_)
        self.destroy_timer(self.timer_)
        return TransitionCallbackReturn.SUCCESS

    def publish_talker(self):
        self.counter_ += 1
        msg = String()
        msg.data = "Hello World: " + str(self.counter_)
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = MyTalkerNode()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        if rclpy.ok():
            node.destroy_node()

if __name__ == "__main__":
    main()

File: ros2_ws/src/labs_cpp_pkg/src/my_talker.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

class MyTalkerNode : public rclcpp::Node
{
public:
  MyTalkerNode() : Node("my_talker"), counter_(1)
  {
    publisher_ = create_publisher<example_interfaces::msg::String>("my_chatter", 10);
    timer_ = create_wall_timer(1s, [this] { publish_talker(); });
  }

private:
  void publish_talker()
  {
    auto msg = example_interfaces::msg::String();
    msg.data = std::string("Hello world: ") + std::to_string(counter_);
    publisher_->publish(msg);
    RCLCPP_INFO(get_logger(), "Publishing: 'Hello World: %d'", counter_);
    counter_++;
  }
  int counter_;
  rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;

};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<MyTalkerNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}